### PR TITLE
extract keyvault AAD resource from vault url

### LINKED ISSUE #398: Add support for getting access tokens on custom environment
**Is this a BUG REPORT or FEATURE REQUEST?**:
 FEATURE REQUEST

**What happened?**:

**What did you expect to happen?**:

**How do you reproduce it (as minimally and precisely as possible)?**:

**Environment**:

**Anything else we need to know?**:

### PR DESCRIPTION
**Pull request by shahzzzam**
**Purpose of the PR:**
Extract keyvault AAD resource from Vault URL.

**Fixes #**
Fixes #398 

Tested it:

```sh
samashah@testacbvm:~$ cat task-secrets.yaml 
version: 1.0-preview-1
secrets:
  - id: mysecret
    akv: https://myacbvault32.vault.azure.net/secrets/SampleSecret/e20cbf6577194b50b3ec013c81c724d3
steps:
  - cmd: bash echo mysecret value is {{.Secrets.mysecret}}


samashah@testacbvm:~$ sudo docker run -v $(pwd):/workspace --workdir /workspace \
> -v /var/run/docker.sock:/var/run/docker.sock acb exec  \
> --homevol $(pwd) -f task-secrets.yaml
2019/03/06 02:14:38 Using /home/samashah as the home volume
2019/03/06 02:14:39 Creating Docker network: acb_default_network, driver: 'bridge'
2019/03/06 02:14:39 Successfully set up Docker network: acb_default_network
2019/03/06 02:14:39 Setting up Docker configuration...
2019/03/06 02:14:39 Successfully set up Docker configuration
2019/03/06 02:14:39 Executing step ID: acb_step_0. Working directory: '', Network: 'acb_default_network'
2019/03/06 02:14:39 Launching container with name: acb_step_0
mysecret value is ACB Secret
2019/03/06 02:14:40 Successfully executed container: acb_step_0
2019/03/06 02:14:40 Step ID: acb_step_0 marked as successful (elapsed time in seconds: 0.671433)
```

### CODE REVIEW

File: cmd/acb/commands/getsecret/getsecret.go [hunk 1]
```diff
@@ -31,27 +30,18 @@ var Command = cli.Command{
 					Name:  "client-id",
 					Usage: "the MSI user assigned identity client ID",
 				},
-				cli.StringFlag{
-					Name:  "vault-resource-url",
```
Comment (ehotinger) [inline cmd/acb/commands/getsecret/getsecret.go]: Update `getsecret_akv.md` too?

File: vaults/azurekeyvault.go [hunk 2]
```diff
@@ -81,8 +81,18 @@ func NewAKVSecretConfig(vaultURL, msiClientID, vaultAADResourceURL string) (*AKV
 		secretVersion = urlSegments[3]
 	}
 
+	vaultHostWithScheme := fmt.Sprintf("%s://%s", parsedURL.Scheme, parsedURL.Host)
```
Comment (mnltejaswini) [inline vaults/azurekeyvault.go]: Why do you need to construct the host with scheme? you can directly split the Host right?

File: vaults/azurekeyvault.go [hunk 3]
```diff
@@ -81,8 +81,18 @@ func NewAKVSecretConfig(vaultURL, msiClientID, vaultAADResourceURL string) (*AKV
...
+	splitStr := strings.SplitAfterN(vaultHostWithScheme, ".", 2)
+	// Ex. https://myacbvault.vault.azure.net -> ["https://myacbvault." "vault.azure.net"]
+	if len(splitStr) != 2 {
+		return nil, fmt.Errorf("extracted vault resource %s from resource ID %s is invalid", vaultHostWithScheme, vaultURL)
```
Comment (mnltejaswini) [inline vaults/azurekeyvault.go]: nit: extracted from vault URL ?

Comment (shahzzzam) [reply vaults/azurekeyvault.go]: Thats because we pass vaultHostWithScheme to `VaultURL` below this line

File: vaults/azurekeyvault.go (hunk 2 above)
Comment (shahzzzam) [reply vaults/azurekeyvault.go]: Thats because we pass vaultHostWithScheme to VaultURL below this line

**Pull request merged**